### PR TITLE
An example how to use an image as a 3d viewport

### PR DESCRIPTION
**Pull request by bekraoui**
This is an example how to use an image as a 3d viewport. remarq: for the now:

- the cube is not rotating (i dont know why yet) it seems like the  is not updating.
- applying padding to the <body> a margin  to the  in the css will cause the cube to drift out of the image.

it is a custumization of the bevy render to texture exemple. with more simplificaiton and a rotationg cube (not rotating yet :) ) inside the viewport.

### CODE REVIEW

**Comment by bekraoui** [comment]
hi, this is a screenshot of the result:

screenshot_0

**Comment by jkb0o** [comment]
Wow! Looks nice! Thanks! I'll check it and leave comments today or tomorrow.

**Comment by bekraoui** [comment]
Thank you for this awesome project. Take your time

File: examples/image-render-to-texture.rs [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+            // render before the "main pass" camera
+            priority: -1,
+            target: RenderTarget::Image(image_handle.clone()),
+            ..default()
+        },
+        transform: Transform::from_translation(Vec3::new(0.0, 0.0, 15.0))
+            .looking_at(Vec3::default(), Vec3::Y),
+        ..default()
+    })
+    .insert(viewport_pass_layer);
```
Comment (jkb0o) [inline examples/image-render-to-texture.rs:145]: I found that the render UICamera doesn't respects the RenderLayer: https://github.com/bevyengine/bevy/issues/6069 and the resulting ui renders with artifacts (if I change background-color for image, for example).

Provided in the issue workaround works like a charm, so my suggestion is to notice the bug in example and add
```rust
.insert(UiCameraConfig { show_ui: false })
```

File: examples/image-render-to-texture.rs [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+	    None        => asset_server.load("icon.png"),
+    };
+    
+    commands.spawn(Camera2dBundle::default());
+    //use some css for the viewport if you want to
+    /*commands.add(StyleSheet::parse("
+            body { padding: 50px; }
+            #viewport { margin: 50px; }
+        ",));*/
+    //set the width and the height of the viewport
```
Comment (jkb0o) [inline examples/image-render-to-texture.rs:166]: I suggest to uncomment stylesheets and provide `background-color` and relative sizes, so user can resize the window and see how rendered widget behaves. For me (after fixing bug from previous comment) it works like a charm. I've tried with this styles:
```css
#viewport {
  margin: 50px; 
  width: 50%; 
  height: 50%; 
  background-color: grey;
}
```

**Review by jkb0o — CHANGES_REQUESTED** [review]
It looks good and will work like a charm after resolving comments.

Also, It would be cool if you do `cargo fmt`

Thank you! And sorry it took so many days!

**Comment by jkb0o** [comment]
And I didn't met issues with cube: it is rotates and doesn't drift.

**Comment by bekraoui** [comment]
The drifting happens when i uncomment the style. I will try it again and make the changes you recomanded as soon as possible (probably in 2 or 3 days)

**Comment by bekraoui** [comment]
it works perfectly. thank you
on the other side cargo fmt does not seem to make any change; by the way i'm new to rust and and developping UI apps so hope to learn from you :).

**Review by jkb0o — APPROVED** [review]

**Comment by jkb0o** [comment]
Thank you!

**Pull request merged**